ui/ozone/platform/wayland/wayland_window.cc

```diff
@@ -363,25 +366,60 @@ void WaylandWindow::ConvertEventLocationToCurrentWindowLocation(
 void WaylandWindow::HandleSurfaceConfigure(int32_t width,
                                            int32_t height,
                                            bool is_maximized,
-                                           bool is_fullscreen) {
+                                           bool is_fullscreen,
+                                           bool is_activated) {
+  // Handle restore state from wayland side: is the state was minimized and
+  // a user restore windows from panel tray, the only way to know that the
+  // window is restored is by checking if the window has been minimized and
+  // |is_activated| is set to true.
+  if (is_minimized_ && is_activated) {
+    is_minimized_ = false;
+    was_minimized_ = !is_minimized_;
```

Comment:
for clarity, I would say was_minimized_ = true here, but your take.

Reply:
fixed